### PR TITLE
Dex 66996 foreign address discrepancy

### LINKED ISSUE #66996: 686 Foreign Base Address Discrepancy
## Issue Description
_Additional validation was added to the downstream VBMS system that is causing null claims if the foreign address flag appears on the claim but the postal code entered does not match the range of foreign base codes._

---
## Tasks
- [ ] Update front-end error messaging based on this [Sketch file](https://www.sketch.com/s/870d13d5-a465-48a3-acc2-32682ba8f780)
- [ ] Add postal code validation. If the foreign address box is selected, the postal code entered must coincide with a APO/FPO/DPO postal code:
```
if (cityname is one of APO/FPO/DPO OR mlty_postal_type_cd is one of AA/AP/AE )
{    if mlty_postal_type_cd is AA, first three characters of zipCode should be 340 else reject transaction if mlty_postal_type_cd is AE, zipCode range should be (090-099) else reject transaction if mlty_postal_type_cd is AP, zipCode range should be (962-966) else reject transaction  }

```

Requirements update: If the Veteran has selected `My mailing address is outside the United States on a U.S. military base` then all 3 zip code validations should be applied. 

## Acceptance Criteria
- [ ] Error messaging appears to user
- [ ] Selecting the foreign address checkbox and entering in a non-APO/DPO/FPO postal code displays an error to the user and prevents submission
---

### PR DESCRIPTION
**Pull request by evansmith**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- Added validation on the zip code field to prevent submission if the "lives on foreign base" field is checked and an invalid zip code is provided. Added additional validation if AE/AP/AA is selected and the zip code does not match the relevant format, as described in the git issue. 
- To replicate the initial issue: When submitting a 686c claim, check the "lives on foreign base" box on the veteran information page and then provide a domestic zip code. 
- Additional validation on the field will prevent submissions with a domestic zip code.
- VA Benefits and Claims
- No flipper

## Related issue(s)

- _[Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#0000](https://github.com/department-of-veterans-affairs/va.gov-team/issues/66996)

## Testing done

- Previously there was no validation on the zip code field outside of confirming it was 5 digits and present. Now there is validation if an invalid zip code is entered when the foreign base box is checked. 
- When filling out a 686c claim, check the box that says: "I receive mail outside of the United States on a U.S. military base." Choose APO and then choose states. The following regex must be followed for each state: AE (^0{1}9{1}[0-9]{3}) AP (^9{1}6{1}[2-6]{1}[0-9]{2}) AA (^3{1}4{1}0{1}[0-9]{2}). If APO/DPO/FPO is chosen without a state and a zip code is entered that does not conform to any of the three regex's above, the error will state: `This postal code is within the U.S. If your mailing address is in the U.S., uncheck the checkbox “I receive mail outside of the United States on a U.S. military base”. If your mailing address is an AFO/FPO/DPO address, enter the postal code for the military base.` If AE/AA/AP is chosen and an invalid zip code is used the error will state: `Because your address is outside the United States on a military base, please provide an APO/FPO/DPO postal code.`
- Tested with each State and relevant regex format. Updated Unit Tests.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Screenshot 2023-10-16 at 11 23 58 AM |Screenshot 2023-10-16 at 11 27 01 AM|
| Desktop | Screenshot 2023-10-16 at 11 23 41 AM| Screenshot 2023-10-16 at 11 26 52 AM Screenshot 2023-10-16 at 11 26 55 AM|

## What areas of the site does it impact?

686c-674 Form including veteran information, stepchild address, add-child additional information, spouse address, and student address.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by tblackwe — CHANGES_REQUESTED** [review]
in the vicinity of line 88 of `.github/CODEOWNERS` can you add

`# Dependent Benefits Experience`
`src/applications/disability-benefits/686c-674 @department-of-veterans-affairs/va-platform-cop-frontend @department-of-veterans-affairs/benefits-dependents-management`

**Comment by tblackwe** [comment]
@Midge-dev @DLarson-Oddball Can you give me a hand with reviewing these front end tickets? 
@tfink419 as well for visibility

**Review by tblackwe — APPROVED** [review]

File: src/applications/disability-benefits/686c-674/config/address-schema.js
Comment (DLarson-Oddball) [inline src/applications/disability-benefits/686c-674/config/address-schema.js:526]: Case is different here than in validationKeys

File: src/applications/disability-benefits/686c-674/config/address-schema.js
Comment (DLarson-Oddball) [inline src/applications/disability-benefits/686c-674/config/address-schema.js:543]: NP: Could we assign the two .some call results to variables then check them explicitly vs using Object.values? 

This currently checks all values in the hash, so a code update in the future might have some unexpected side effects if the future developer isn't aware this check exists.

**Review by DLarson-Oddball — DISMISSED** [review]
Noticed one case mismatch, and included an optional nitpick. Specs are excellent!

File: src/applications/disability-benefits/686c-674/config/address-schema.js
Comment (evansmith) [reply src/applications/disability-benefits/686c-674/config/address-schema.js:543]: @DLarson-Oddball the hash also has the default vals of base, address, and childAddressInfo. I figured this check with Object.values is just to see whether we even need to run the logic/validations.

File: src/applications/disability-benefits/686c-674/config/address-schema.js
Comment (evansmith) [reply src/applications/disability-benefits/686c-674/config/address-schema.js:526]: great catch!

**Review by tblackwe — DISMISSED** [review]

File: src/applications/disability-benefits/686c-674/config/address-schema.js
Comment (tfink419) [reply src/applications/disability-benefits/686c-674/config/address-schema.js:543]: I agree, if any additional value is added to the api response that equals `true` it would also cause this to be true.

**Review by tfink419 — DISMISSED** [review]

**Review by Midge-dev — DISMISSED** [review]

**Review by Midge-dev — APPROVED** [review]
LGTM

**Review by tblackwe — APPROVED** [review]

File: .github/CODEOWNERS
Comment (CBonade) [inline .github/CODEOWNERS:90]: One of the teams listed here is flagged as not existing.

**Review by CBonade — CHANGES_REQUESTED** [review]
Just seeing one error flagged by GitHub here about codeowners being specified for a group that does not exist. Please correct this so we can avoid having errors in our codeowners file globally, or ensure the group is created to eliminate this before merging.

File: .github/CODEOWNERS
Comment (tblackwe) [reply .github/CODEOWNERS:90]: the team exists, does something need to be done to grant it visibility in vets-website?
https://github.com/orgs/department-of-veterans-affairs/teams/benefits-dependents-management

Comment (CBonade) [reply .github/CODEOWNERS:90]: This could also be possibly caused by the team not having write access to the vets-website repository

**Review by CBonade — APPROVED** [review]

File: .github/CODEOWNERS
Comment (CBonade) [reply .github/CODEOWNERS:90]: I've added write access for your group and this has cleared it up. Approved PR!

**Pull request merged**
